src/main/java/com/programmers/commandline/domain/voucher/repository/impl/MemoryVoucherRepository.java

```diff
@@ -0,0 +1,41 @@
+package com.programmers.commandline.domain.voucher.repository.impl;
+
+import com.programmers.commandline.domain.voucher.entity.Voucher;
+import com.programmers.commandline.domain.voucher.repository.VoucherRepository;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+@Profile({"local"})
+public class MemoryVoucherRepository implements VoucherRepository {
+
+    private List<Voucher> memory = new ArrayList<>();
+
+    @Override
+    public String save(Voucher voucher) {
+         memory.add(voucher);
+        return voucher.getId();
+    }
+
+    @Override
+    public List<Voucher> findAll() {
+        return memory;
+    }
+
+    @Override
+    public Optional<Voucher> findById(String voucherId) {
+        Optional<Voucher> voucher = Optional.empty();
+
+        for (Voucher v : memory) {
+            if (v.getId().equals(voucherId)) {
+                voucher = Optional.of(v);
+            }
+        }
+
+        return voucher;
```

Comment:
```suggestion
        return memory.stream()
            .filter(v -> v.getId().equals(voucherId))
            .findFirst();
```
다음과 같이 작성해 볼 수 있을 것 같아요!

Reply:
findFirst()의 반환타입이 Optional이라는 것을 새롭게 알게 되었습니다.🔥